metrics-operator/controllers/common/analysis/objective_evaluator.go

```diff
@@ -0,0 +1,71 @@
+package analysis
+
+import (
+	"errors"
+	"strconv"
+
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1alpha3"
+	"github.com/keptn/lifecycle-toolkit/metrics-operator/controllers/common/analysis/types"
+)
+
+type ObjectiveEvaluator struct {
+	TargetEvaluator ITargetEvaluator
+}
+
+func NewObjectiveEvaluator(t ITargetEvaluator) ObjectiveEvaluator {
+	return ObjectiveEvaluator{
+		TargetEvaluator: t,
+	}
+}
+
+func (oe *ObjectiveEvaluator) Evaluate(values map[string]string, obj *v1alpha3.Objective) types.ObjectiveResult {
+	result := types.ObjectiveResult{
+		Score: 0.0,
+		Value: 0.0,
+	}
+
+	// get the value
+	floatVal, err := getValueFromMap(values, computeKey(obj.AnalysisValueTemplateRef))
+	if err != nil {
+		result.Error = err
```

Comment:
I don't know if this error message makes sense to be stored in the CR 🤔 
looking at the code, it seems that this `.Error` field is not used. Maybe we could propagate the error up so we can log it at the entry point with info about what analysis fails

Reply:
the error is filled when the map of values does not contain the required key and value, therefore we store the error that the value for the objective was not retrieved and pass the result one layer above. Therefore in the end, each `ObjectiveResult` has an error stored in the structure if it occured

Reply:
The struct is only internally used and it is not stored in any CR. However, the field it is not used so I am questioning the utility of storing errors. Furthermore, we are going against Go practices to have the error flow as part of the function return type and instead, we hide that into the returning struct :/